cpp/src/bitmask/null_mask.cu

```diff
@@ -430,25 +403,20 @@ void inplace_bitmask_and(bitmask_type *dest_mask,
 {
   CUDF_EXPECTS(std::all_of(begin_bits.begin(), begin_bits.end(), [](auto b) { return b >= 0; }),
                "Invalid range.");
-  CUDF_EXPECTS(mask_size > 0, "Invalid bit range.");
   CUDF_EXPECTS(std::all_of(masks.begin(), masks.end(), [](auto p) { return p != nullptr; }),
                "Mask pointer cannot be null");
 
-  auto number_of_mask_words = num_bitmask_words(mask_size);
-
   rmm::device_vector<bitmask_type const *> d_masks(masks);
   rmm::device_vector<size_type> d_begin_bits(begin_bits);
 
-  cudf::detail::grid_1d config(number_of_mask_words, 256);
-  offset_bitmask_and<<<config.num_blocks, config.num_threads_per_block, 0, stream.value()>>>(
-    dest_mask,
-    d_masks.data().get(),
-    d_begin_bits.data().get(),
-    d_masks.size(),
+  inplace_bitmask_binop(
+    [] __device__(bitmask_type left, bitmask_type right) { return left & right; },
+    device_span<bitmask_type>(dest_mask, num_bitmask_words(mask_size)),
+    device_span<bitmask_type const *>(d_masks.data().get(), d_masks.size()),
+    device_span<size_type>(d_begin_bits.data().get(), d_begin_bits.size()),
     mask_size,
-    number_of_mask_words);
-
-  CHECK_CUDA(stream.value());
+    stream,
+    mr);
```

Comment:
Do we also need a `stream.synchronize();` here?

Reply:
Yes, good catch.